### DISCUSSION TITLE
Slow performance in windows 10 start menu on windows 11

### DISCUSSION THREAD

**Discussion by ypgrimm** [valinet/ExplorerPatcher#2571] [Q&A]
Although it's not extremely important the fps and responsiveness is really bad on the windows 10 start menu and taskbar. Is this is a general issue or am I doing something wrong.

**Comment by ypgrimm** [comment]
Version 22H2 (OS Build 22621.2715)

**Comment by pyrates999** [comment]
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.

Please include any other software you have installed that modifies the UI of windows.

**Comment by ypgrimm** [comment]
Version 22621.2506.60.1

I have the Windows 11 taskbar (I tried the Windows 10 taskbar but issue persisted), the Windows 10 start menu, and disabled the windows+f bind.

I have lively wallpaper if that counts but nothing else.

**Comment by pyrates999** [comment]
Try the latest prerelease: https://github.com/valinet/ExplorerPatcher/releases/tag/22621.2506.60.1_a7a3d27

**Reply by ypgrimm** [reply]
I just installed it but when scrolling apps on start menu it still feels choppy, maybe its placebo

**Reply by pyrates999** [reply]
take a video and compare between when EP is installed and when EP is not installed.

here's how to do that: https://www.zdnet.com/article/how-to-screen-record-in-windows-10-or-windows-11/

**Reply by ypgrimm** [reply]
I can't because it's the windows 10 start menu and I'm on 11, I don't have the same issue on the windows 11 one.

**Comment by Amrsatrio** [comment]
It's fine on my devices 🤷 Of course when I have no games open. But when I have a game open it lags, and I believe it's normal.

- i7-8550U, 2x8GB DDR3, UHD 620, Windows 11 Pro 22631.2787
- i9-12900K, 2x32GB DDR5, 3090 Ti, Windows 11 Pro 22621.2715

**Reply by ypgrimm** [reply]
Scrolling on the apps list just feels choppy (or lower than my monitor refresh rate of 240hz) I don't know why, it must not be ep then, thanks for helping.

**Comment by stamminator** [comment]
@Amrsatrio I'm having the same issue. Here are some screen recordings of the issue, at 60fps so the choppiness is more obvious. I have a pretty powerful machine and never have smoothness issues with other parts of the OS (including ExplorerPatcher's taskbar) running at 100hz, so it's definitely an issue with the Windows 10 start menu implementation.

#### Scrolling:
At first glance, it looks like part of the issue may be inefficient virtual scrolling. See how the icons being scrolled into view need to reload, even the ones I just scrolled past a couple of seconds earlier?

![scrolling issue](https://github.com/user-attachments/assets/ff51883f-ebcf-4010-8cf1-e4f6b7f49fd4)

#### Jump lists (context menu):
This one is definitely related to the number of items in the jump list. For apps with no options beyond the four default Unpin, Resize, More, and Uninstall options (e.g. the ExplorerPatcher Properties app), the delay in between the right-click and the jump list's appearance is around 200ms. For an app with just a few more options (e.g. Calculator), the delay is around 600ms, and the choppiness when hovering the cursor over those options starts to happen if you look closely. But for an app with a maxed out jump list (e.g. Visual Studio with 12 items in the "Recent" list), the delay can be almost 2 seconds, and the hover choppiness feels like I'm running a Pentium II with no thermal paste. In fact, when right clicking the Visual Studio 2017 icon in my recording, it even froze my ffmpeg screen recorder for a moment, which is why it skips for a moment. Seems to be some thread blocking going on.

![jump list issue](https://github.com/user-attachments/assets/ff6bf88e-a901-402b-be91-11b772e34099)

ExplorerPatcher version 22621.3810.66.1
Windows 11 build 22631.3737
[DxDiag.txt](https://github.com/user-attachments/files/16184680/DxDiag.txt)

**Reply by Amrsatrio** [reply]
This might be due to how EP fixes the context menu. It hijacks the resource loader to prevent the new context menu styles from being loaded. That means every resource load is scanned by the patcher, thereby causing the lags. Also the lags may be because of the number of tiles you have. Nonetheless I do not experience that extreme of lags on my systems, so I find it weird.

**Reply by stamminator** [reply]
After a system reboot, I'm getting pretty smooth performance as well. I'm going to pay attention throughout the day to see if/when it changes.

**Reply by pyrates999** [reply]
Was the system reboot due to installing windows updates?

**Reply by stamminator** [reply]
> Was the system reboot due to installing windows updates?

No, just my morning reboot. And as of this morning's reboot, it's slow again. I wonder if there's a race condition for something starting up. I'm going to experiment with some different startup configurations and disabling/enabling certain apps to see if I can isolate the fail condition.

**Reply by pyrates999** [reply]
Do you have any other software installed that modifies the UI of windows?

**Reply by stamminator** [reply]
I believe I've isolated the issue. The bad performance takes place when hardware-accelerated GPU scheduling is enabled. After disabling it, it's smooth.

![hE4CDM1X6Y](https://github.com/user-attachments/assets/a7d6fbfc-ea5e-4975-b578-beccac97c6d2)

If anyone else is able to reproduce this as the cause, it might be worth mentioning in #1679.

**Reply by pyrates999** [reply]
are your gpu drivers up to date?

**Reply by stamminator** [reply]
I was nearly up-to-date with my RTX 3050 Ti laptop WHQL driver 555.99 (Jun 04, 2024), but after installing the latest WHQL driver 560.70 which literally dropped yesterday, it seems to be working fine with hardware accelerated GPU scheduling enabled, even after several reboots.

My advice would be to add a section to #1679 stating that if start menu performance is bad, it's likely a graphics driver issue and that after updating your driver, if the issue is still taking place, hardware accelerated GPU scheduling should be disabled.

@ypgrimm Are you still experiencing the issue? If so, can you try these steps?

**Reply by stamminator** [reply]
Well damn. The issue started happening again, although I've made no driver or graphics settings changes. I've even tried experimenting back and forth with setting everything I can think of (`explorer.exe`, `dwm.exe`, `ep_dwm.exe`, `ShellExperienceHost.exe`, and `StartMenuExperienceHost.exe`) to explicitly use either the iGPU or the dGPU... 

![B4KFQi8Uaz](https://github.com/user-attachments/assets/32421a8b-94b2-470e-8eb9-0d887fb633f5) 

...but as long as hardware-accelerated GPU scheduling (HAGS) is enabled, I seem to have this problem intermittently and can't determine why it sometimes runs fine even with HAGS turned on.

The only time I seem to avoid the issue 100% of the time is if HAGS is disabled.

To make troubleshooting easier, can you suggest a way that I can determine with certainty which GPU the start menu is using? Right now I'm just pressing buttons and making an educated guess. If I can be sure, I may be able to isolate the issue.

**Reply by pyrates999** [reply]
This looks like an issue with the GPU drivers and windows itself.

**Reply by stamminator** [reply]
@pyrates999 Yeah, I give up. This morning, I saw the issue happening even though HAGS is disabled. Unless anyone has advice for some lower level troubleshooting, I'll just have to put up with the intermittent bad performance. Sorry to clutter up the discussion with theories that didn't pan out.

**Reply by pyrates999** [reply]
no worries.
